src/main/java/org/folio/roles/controller/RoleCapabilityController.java

```diff
@@ -34,6 +36,20 @@ public ResponseEntity<RoleCapabilities> createRoleCapabilities(RoleCapabilitiesR
       .totalRecords(pageResult.getTotalRecords()));
   }
 
+  @Override
+  public ResponseEntity<RoleCapabilities> createRoleCapabilitiesByPermissionNames(
+    RolePermissionNamesRequest rolePermissionNamesRequest) {
+    var capabilityIds = capabilityService.findByPermissionNames(rolePermissionNamesRequest.getPermissionNames())
+      .stream()
+      .map(Capability::getId)
+      .toList();
+    var roleCapabilitiesRequest = new RoleCapabilitiesRequest()
+      .roleId(rolePermissionNamesRequest.getRoleId())
+      .capabilityIds(capabilityIds);
```

Comment:
this does not belong to the controller class and must be placed at service layer